crossbar/controller/node.py

```diff
@@ -365,6 +364,21 @@ def _run_node_config(self, config):
 
                     component_no = 1
 
+                    # if components exit "very soon after" we try to
+                    # start them, we consider that a failure and shut
+                    # our node down. We remove this subscription 2
+                    # seconds after we're done starting everything
+                    # (see below). This is necessary as
+                    # start_container_component returns as soon as
+                    # we've established a connection to the component
+                    def component_exited(info):
+                        dead_comp = info['id']
+                        log.msg("Component '{}' failed to start; shutting down node.".format(dead_comp))
+                        if self._reactor.running:
+                            self._reactor.stop()
+                    topic = 'crossbar.node.{}.worker.{}.container.on_component_stop'.format(self._node_id, worker_id)
+                    component_stop_sub = yield self._controller.subscribe(component_exited, topic)
+
                     for component in worker.get('components', []):
```

Comment:
This smells a bit hacky, but handles any case where the component dies very shortly after starting...